tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/RemoveInternetSourcesFromNuGetConfig.cs

```diff
@@ -31,7 +33,14 @@ public override bool Execute()
             {
                 if (e.Name == "add")
                 {
-                    return !(e.Attribute("value").Value.StartsWith("http://", StringComparison.OrdinalIgnoreCase) || e.Attribute("value").Value.StartsWith("https://", StringComparison.OrdinalIgnoreCase));
+                    if (OfflineBuild)
+                    {
+                        return !(e.Attribute("value").Value.StartsWith("http://", StringComparison.OrdinalIgnoreCase) || e.Attribute("value").Value.StartsWith("https://", StringComparison.OrdinalIgnoreCase));
+                    }
+                    else
+                    {
+                        return !(e.Attribute("value").Value.StartsWith("https://pkgs.dev.azure.com/dnceng/_packaging", StringComparison.OrdinalIgnoreCase));
```

Comment:
Factor out `e.Attribute("value").Value.` to make both branches easier to read?

Reply:
Good idea, done.